types/block.go

```diff
@@ -68,3 +74,166 @@ type Signature []byte
 type IntermediateStateRoots struct {
 	RawRootsList [][]byte
 }
+
+func (h *Header) ToProto() *pb.Header {
+	return &pb.Header{
+		Version: &pb.Version{
+			Block: h.Version.Block,
+			App:   h.Version.App,
+		},
+		NamespaceId:     h.NamespaceID[:],
+		Height:          h.Height,
+		Time:            h.Time,
+		LastHeaderHash:  h.LastHeaderHash[:],
+		LastCommitHash:  h.LastCommitHash[:],
+		DataHash:        h.DataHash[:],
+		ConsensusHash:   h.ConsensusHash[:],
+		AppHash:         h.AppHash[:],
+		LastResultsHash: h.LastResultsHash[:],
+		ProposerAddress: h.ProposerAddress[:],
+	}
+}
+
+func (h *Header) FromProto(other *pb.Header) error {
+	h.Version.Block = other.Version.Block
+	h.Version.App = other.Version.App
+	if !safeCopy(h.NamespaceID[:], other.NamespaceId) {
+		return errors.New("invalid lenght of 'NamespaceId'")
+	}
+	h.Height = other.Height
+	h.Time = other.Time
+	if !safeCopy(h.LastHeaderHash[:], other.LastHeaderHash) {
+		return errors.New("invalid lenght of 'InvalidLastHeaderHash'")
```

Comment:
Typo here and in other cases
```suggestion
		return errors.New("invalid length of 'InvalidLastHeaderHash'")
```

Reply:
OMG :facepalm: